src/Common/Commands.Common/Extensions/CmdletExtensions.cs

```diff
@@ -178,6 +178,30 @@ public static void SafeCopyValue<T>(this FieldInfo field, T source, T target)
             }
         }
 
+        /// <summary>
+        /// Safely copy the selected parameter set from one cmdlet to another
+        /// </summary>
+        /// <typeparam name="T">The cmdlet type</typeparam>
+        /// <param name="source">The cmdlet to copy the parameter set name from</param>
+        /// <param name="target">The cmdlet to copy to</param>
+        public static void SafeCopyParameterSet<T>(this T source, T target) where T: AzurePSCmdlet
+        {
+            if (source != null && target != null)
+            {
+                if (!string.IsNullOrWhiteSpace(source.ParameterSetName))
+                {
+                    try
+                    {
+                        target.SetParameterSet(source.ParameterSetName);
+                    }
+                    catch
+                    {
+
+                    }
```

Comment:
would be good to at least put some info somewhere (in the debug stream?) that there were an exception.

Reply:
See this PR https://github.com/Azure/azure-powershell/pull/5086 @markcowl

Reply:
This is bootstrapping code that cannot fail, so we should not use the PowerShell ICommandRuntime